src/components/select/select-list/select-list.component.js

```diff
@@ -351,24 +401,37 @@ const SelectList = React.forwardRef(
       const indexOfMatch = getIndexOfMatch(highlightedValue);
 
       setCurrentOptionsListIndex(indexOfMatch);
-      updateListScrollTop(
-        indexOfMatch,
-        multiColumn ? tableRef.current : listRef.current,
-        optionRefList
-      );
+
+      virtualizer.scrollToIndex(indexOfMatch, SCROLL_OPTIONS);
+      // TODO: is there a better way than calling handleListScroll manually?
+      handleListScroll({ target: listContainerRef.current });
     }, [
-      childrenList,
       getIndexOfMatch,
       highlightedValue,
-      multiColumn,
-      optionRefList,
+      virtualizer,
+      handleListScroll,
+      listContainerRef,
```

Comment:
Apologies if I missing something, but does having the ref in the dependency array do anything here, I thought mutating refs do not trigger re-renders etc?

Reply:
This is more just to keep the linter happy - I know missing dependencies are warnings rather than errors but I still prefer to avoid them.

Although there is a potential need - this is a forwarded ref rather than one created using `useRef`, it's possible that components that consume this pass a `ref` that's conditional and can change from one object to a completely different one, or something weird like that. It likely doesn't matter in practice but I think it's best to keep it in the dependencies. (It's always safer in React to have too many dependencies than not enough - a missing dependency is a bug waiting to happen, while an unnecessary one ought to only cause performance degradations at worst.)

Reply:
No worries thanks for clarifying 👍